src/random.cr

```diff
@@ -277,6 +290,113 @@ module Random
     end
   end
 
+  # Fills a given slice with random bytes.
+  #
+  # ```
+  # slice = Bytes.new(4) # => [0, 0, 0, 0]
+  # Random.new.random_bytes(slice)
+  # slice # => [217, 118, 38, 196]
+  # ```
+  def random_bytes(buf : Bytes) : Nil
+    ptr = buf.to_unsafe
+    finish = buf.to_unsafe + buf.size
+
+    while ptr < finish
+      random = next_u
+      rand_ptr = pointerof(random).as(UInt8*)
+      if IO::ByteFormat::SystemEndian != IO::ByteFormat::LittleEndian
```

Comment:
Is it needed for anything except tests?

Reply:
This is needed to produce the same results for the same seed, on different systems.
Note that this code is NOT only used for `Random::System`